plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -1072,6 +1047,39 @@ public function render_order_status_column( WC_Order $order ): void {
 		}
 	}
 
+	/**
+	 * Gets the order status label for an order.
+	 *
+	 * @param WC_Order $order The order object
+	 *
+	 * @return void
+	 */
+	private function get_order_status_label( WC_Order $order ): string {
+		$status_names = array(
+			"Pending payment" => "The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action.",
+			"On hold" => "The order is awaiting payment confirmation. Stock is reduced, but you need to confirm payment.",
+			"Processing" => "Payment has been received (paid), and the stock has been reduced. The order is awaiting fulfillment.",
+			"Completed" => "Order fulfilled and complete.",
+			"Failed" => "The customer’s payment failed or was declined, and no payment has been successfully made.",
+			"Draft" => "Draft orders are created when customers start the checkout process while the block version of the checkout is in place.",
+			"Canceled" => "The order was canceled by an admin or the customer.",
+			"Refunded" => "Orders are automatically put in the Refunded status when an admin or shop manager has fully refunded the order’s value after payment."
+		);
+
+		/**
+		 * Provides an opportunity to modify and extend the order status labels.
+		 *
+		 * @param array    $action Order actions.
+		 * @param WC_Order $order  Current order object.
+		 * @since 6.7.0
+		 */
+		$status_names = apply_filters( 'woocommerce_get_order_status_labels', $status_names );
```

Comment:
Nice idea for the filter! Let's aim for the next available release, so the since tag should be `@since 9.1.0`

Could you also please wrap the strings in the `__()` [translation function](https://codex.wordpress.org/I18n_for_WordPress_Developers)? The lint check will also need single quotes. e.g. 

```
__( 'The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action.', 'woocommerce' );
```

Reply:
Thank you @adrianduffell! 

I've made the requested changes, take a look at them whenever you have the time :)